### DISCUSSION TITLE
forge new canvas feedback

### DISCUSSION THREAD

**Discussion by wzwow** [lllyasviel/stable-diffusion-webui-forge#1683] [General]
1. possible to change the new canvas checker background something darker or plain color? it is so distracting when see in dark mode. 
2. zoom function to a slider instead of mouse scroll. it is too easily to zoom the canvas while trying to scroll down the page.

![image](https://github.com/user-attachments/assets/98014136-ffc1-4dad-a606-a0171dfa47a7)

**Comment by mold0101** [comment]
Agree. A delete function would be very helpful too.

**Reply by wzwow** [reply]
i dig in a little into the canvas folder... there is a css and one can edit it. i block 3 line of code and got a blank canvas now  ;D

![image](https://github.com/user-attachments/assets/bb5db4ea-dc7e-4d6e-9070-7a6b44305994)


![image](https://github.com/user-attachments/assets/e9b6c83c-7030-4fb7-9f87-5c1d6771086d)

**Comment by pflky** [comment]
I just add this to styles.css. Also increases the size of the mouseover toolbar. Of course updates can erase this, so I keep a backup of changes to re-add when that happens. This covers both controlnet and img2img windows. I didn't account for img2img changes though, so the toolbar will be a little clipped in certain tabs like the sketch tab if you use the toolbar size changes.

}
.forge-image-container {
background-image: none !important;
background-color: #0b0f19 !important;
}
.forge-toolbar {
transform: scale(1.7);
top: 20px;
left: 80px;
}

![Screenshot_1](https://github.com/user-attachments/assets/c8c18471-0c20-4e7e-9c2a-527a7e6d8976)

**Comment by Jengall** [comment]
I like the mouse scroll. But to prevent what you don't like on A1111 they use the shift key with the scroll mouse. It would be great.
I would love to have the brush size on ctrl+mouse scroll too.
And a mask eraser would be awesome too

**Comment by E2GO** [comment]
Also did some feedback on it here: https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/853#discussioncomment-10408417

1. Add an option to enable or disable mouse scroll on the inpaint window. Reason: When the mouse cursor is over the inpaint window and you try to scroll the Forge, you might accidentally zoom in or out of the inpaint window.
2. Separate the controls for resetting the mask editing and resetting the view, or add an enable/disable option for this feature. Now - clearing mask also resets window.
3. Introduce shortcuts for resizing the brush size, such as the [ ] keys, similar to Photoshop. This would be particularly useful when using a tablet.
4. Allow erasing already painted mask by holding down the ALT key.
5. Enable the ability to continue drawing the mask even when the cursor moves outside the canvas, preventing interruptions when the mouse goes out of the image.

**Comment by Kmolpat** [comment]
Using a pen doesn't work. When I try to draw, the canvas just drags.

**Comment by MackinationsAi** [comment]
![Screenshot 2024-11-27 at 00-08-43 Stable Diffusion](https://github.com/user-attachments/assets/f97a85f0-0a4e-4136-a343-5a602ecfc3bb)

![Screenshot 2024-11-26 at 23-53-44 Stable Diffusion](https://github.com/user-attachments/assets/fad628fb-a46b-4313-adcd-a099c9d9d3b8)

Here's a dark_mode version of the new forge canvas feature.`C:\your_forge_path_here\stable-diffusion-webui-forge\modules_forge\forge_canvas`simply replace the files w/ the files in the attached [forge_canvas_dark-mode.zip](https://github.com/user-attachments/files/17928911/forge_canvas_dark-mode.zip) in your forge repo.

**Comment by seowner** [comment]
I was only excited to try Forge because I saw Gradio 4, which has the ability to erase (i.e., manually alter/remove part of the mask using a separate brush) in the canvas (instead of having to undo a million things to change something small)... but somehow it's not here even though supported. Sad -_- I hope it's added soon.

**Reply by wzwow** [reply]
i was secretly hoping to get the mj style free marquee selection tool instead of a paint brush.

**Reply by Haoming02** [reply]
Forge is not using Gradio's Canvas

They have their own custom [forge_canvas](https://github.com/lllyasviel/stable-diffusion-webui-forge/tree/main/modules_forge/forge_canvas)

**Reply by wzwow** [reply]
not sure about what Gradio Canvas can do, but i believe he has his reason to create the forge canvas

**Reply by MackinationsAi** [reply]
The new mask editor in comfy is actually really good, finally got it's first refresh since comfy was released! [Check it out](https://github.com/Comfy-Org/ComfyUI_frontend/pull/1284).

**Comment by MrLawli3t** [comment]
I made an eraser extension if people are interested: https://github.com/MrLawli3t/ForgeUI-MaskEraser-Extension

**Reply by Jengall** [reply]
Damn! I switched on ComfyUI 2 weeks ago just because of that, and 1year later, you solved it! (I lost hope....)
Thx a lot!

**Reply by MrLawli3t** [reply]
Glad I could help!

**Comment by Hereticatch33** [comment]
is there a list of all the hotkeys and functions? really need a eye dropper to sample colors form the img.
